utils/_decorators.py

```diff
@@ -106,109 +94,52 @@ def _should_skip(condition=None, library=None, weblog_variant=None):
     return True
 
 
-def missing_feature(condition: bool = None, library=None, weblog_variant=None, reason=None, force_skip: bool = False):
-    """decorator, allow to mark a test function/class as missing"""
-
-    skip = _should_skip(library=library, weblog_variant=weblog_variant, condition=condition)
-
-    def decorator(function_or_class):
-
-        if inspect.isclass(function_or_class):
-            assert condition is not None or (library is None and weblog_variant is None), _MANIFEST_ERROR_MESSAGE
-
-        if not skip:
-            return function_or_class
-
-        full_reason = "missing_feature" if reason is None else f"missing_feature ({reason})"
-
-        return _get_expected_failure_item(function_or_class, full_reason, force_skip=force_skip)
+def decorator(expected_to_fail, force_skip, condition, decorator_type, reason, function_or_class):
+    if inspect.isclass(function_or_class):
+        assert condition is not None, _MANIFEST_ERROR_MESSAGE
 
-    return decorator
-
-
-def incomplete_test_app(
-    condition: bool = None, library=None, weblog_variant=None, reason=None, force_skip: bool = False
-):
-    """decorator, marks tests that rely on endpoints in test apps that are incomplete or missing"""
-
-    skip = _should_skip(library=library, weblog_variant=weblog_variant, condition=condition)
-
-    def decorator(function_or_class):
+    if decorator_type in ("bug", "flaky"):
+        _ensure_jira_ticket_as_reason(function_or_class, reason)
 
-        if inspect.isclass(function_or_class):
-            assert condition is not None or (library is None and weblog_variant is None), _MANIFEST_ERROR_MESSAGE
+    full_reason = decorator_type if reason is None else f"{decorator_type} ({reason})"
+    if not expected_to_fail:
+        return function_or_class
+    return _get_expected_failure_item(function_or_class, full_reason, force_skip)
 
-        if not skip:
-            return function_or_class
 
-        full_reason = "incomplete_test_app" if reason is None else f"incomplete_test_app ({reason})"
+def missing_feature(condition=None, library=None, weblog_variant=None, reason=None, force_skip: bool = False):
+    """decorator, allow to mark a test function/class as missing"""
+    expected_to_fail = _expected_to_fail(library=library, weblog_variant=weblog_variant, condition=condition)
+    return partial(decorator, expected_to_fail, force_skip, condition, "missing_feature", reason)
```

Comment:
Using named argument will be easier to maintain 

```suggestion
    return partial(decorator, expected_to_fail=expected_to_fail, force_skip=force_skip, condition=condition, decorator_type="missing_feature", reason=reason)
```

Reply:
This fails:

```
from functools import partial 

def method(arg1, arg2, arg3):
    print(f"arg1 is [{arg1}]")
    print(f"arg2 is [{arg2}]")
    print(f"arg3 is [{arg3}]")

x = partial(method, arg2="I'm arg2", arg1="I'm arg1")

x("I'm arg3") # but x(arg3="I'm arg3") would work
```

Reply:
I't because you gave `arg1` in the partial, and also in the call of `x`, so python does not know what is the value for `arg1`. But if there is no ambiguity, it's fine : 

``` python
from functools import partial 

def method(arg1, arg2, arg3):
    print(f"arg1 is [{arg1}]")
    print(f"arg2 is [{arg2}]")
    print(f"arg2 is [{arg3}]")

x = partial(method, arg3="I'm arg3", arg2="I'm arg2")

x("I'm arg1")
```

``` bash
> python tt.py 
arg1 is [I'm arg1]
arg2 is [I'm arg2]
arg2 is [I'm arg3]
```

Th key is that arguments you do not give to `partial`, and won't be passed as named argument must be at the begining of the argument list. It's not working here because the argument not provided to `partial` is `function_or_class`, and it's the last in the list. If you put it in first, it works fine : 

```python
def _decorator(function_or_class, marker, decorator_type, condition, library, weblog_variant, reason):
```

If you don't mind, I'd like to have this, I had very bad experience with positional arguments and `partial`, as it's very easy to mess up.

To be sure I'm not saying crap, I've tested locally. So if you're ok with my suggestion, I can push the commit, just let me know 😄

Reply:
The pytest decorator is called with named arguments so `function_or_class` must be the first argument in the partial (if we plan to call pass the other arguments as keyword arguments). That makes sense. I don't have a preference either way (positional vs keyword), I'll make the change.